### DISCUSSION TITLE
While install chromium: System.IO.FileNotFoundException: 'CefSharp.Core, Version=51.0.0.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138'The system cannot find the file specified

### DISCUSSION THREAD

**Discussion by rachelYedatop** [cefsharp/CefSharp#4116] [Q&A] [answered]
I have a winform projecct in visual studio.
Try to install chromiun to allowed run the projct on chrome and not Internet Explorer.
But after i go ahed : [this article](https://docs.telerik.com/devtools/winforms/knowledge-base/how-to-embed-chrome-browser-in-a-winforms-application), I got this error:
**System.IO.FileNotFoundException: 'לא היתה אפשרות לטעון קובץ או הרכבה 'CefSharp.Core, Version=51.0.0.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138' או אחת מהתלויות שלו. The system cannot find the file specified**

Why? What can I do? 

Thanks!

**Comment by amaitland** [comment, marked as answer]
The official getting started guide is at https://github.com/cefsharp/CefSharp/wiki/Quick-Start with a working reference example provided at https://github.com/cefsharp/CefSharp.MinimalExample

> But after i go ahed : [this article](https://docs.telerik.com/devtools/winforms/knowledge-base/how-to-embed-chrome-browser-in-a-winforms-application), I got this error:

This article is not an official CefSharp guide and is 4 years out of date. Revert the changes you've made and use the quick start guide.

The current supported version 101.0.108, version 51 is ancient and no longer supported.

**Reply by rachelYedatop** [reply]
Thamk you @amaitland I'll try you'r suggestion and will update.

**Comment by rachelYedatop** [comment]
Hi @amaitland, TNX, I was able to add CefSharp according to your instructions :)
But I have a problem now, when I run the app, it clears the cookies every time and does not save the password and the username.
Unlike IE this name saved my data if I did not delete cookies.
Maybe do you know how to resolve it?

**Reply by amaitland** [reply]
An in memory cache is used by default. You need to specify a CachePath to persist cookies etc. See https://github.com/cefsharp/CefSharp/wiki/General-Usage#need-to-knowlimitations

**Reply by rachelYedatop** [reply]
Hi, 
I ran into another problem:
In my computer the app is running well.
but when I try to run the app in a new computer the app is not running, with this error:
`System.IO.FileNotFoundException at CefSharp.CefSettingsBase..ctor() at CefSharp.WinForms.CefSettings..ctor() at PinPadDriverWithCS.Form1.InitializeComponent() at PinPadDriverWithCS.Form1..ctor() at PinPadDriverWithCS.Program.Main() `
So -
When I download and install VC_redistx86.exe the app is running well.
my problem is - since i need to copy the program to a lot of machines' and I don't want to download and install the VC_redistx86.exe in each one - I want to fount solution to assimilate VC_redistx86.exe in the app or in the app folder.
**Is there a solution? DYN?**

**Reply by amaitland** [reply]
Visual C++ 2019 or greater is required. Using the Microsoft provide installer is the recommend approach and what I'd suggest you use. Include the installer with your application. You could have it launch on first run of your application. Or use an installer like WiX (I prefer wixsharp).

You can bin deploy visual c++, https://github.com/cefsharp/CefSharp/wiki/Frequently-asked-questions#Including_vcredist has some guidance. If you have problems then use the installer.

**Reply by rachelYedatop** [reply]
Hi @amaitland,
Thanks for all the help.
I tried the C # side to connect the Angular js (Angular1) side.
But it does not work for me. Maybe because of Js side? (to function: `closeApp()`)
C#:
 ```
browser.JavascriptObjectRepository.Register("jsObj", new ScriptManager(this), isAsync: false, options: BindingOptions.DefaultBinder);
 this.Controls.Add(browser);
 browser.LoadUrl("https://.........");
```
ScriptManager:
```
 private Form1 mForm;
  public ScriptManager(Form1 form)
  {
      mForm = form;
  }
```
closeApp  - the specific function:
```
public void closeApp()
{
    mForm.closeApp();
}
```
Js side:
`window.external.closeApp();` or `window.closeApp();`

**Reply by amaitland** [reply]
Please start a new discussion if you have a new question to ask.

**Comment by rachelYedatop** [comment]
Thank you Alex!
You helped me very, very much 👌👍💪



--
רחל ברוורמן                Rachel Braverman
מחלקת פיתוח   Development Departmen




[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
05/25/22,
07:58:05 PM

‫בתאריך יום ג׳, 24 במאי 2022 ב-22:18 מאת ‪Alex Maitland‬‏ <‪
***@***.***‬‏>:‬
